addons/debuginfo/debuginfo.lua

```diff
@@ -0,0 +1,59 @@
+_addon.name = 'debuginfo'
+_addon.author = 'Lili'
+_addon.version = '1'
+_addon.command = 'debuginfo'
+
+require('logger')
+
+local       get_dir = windower.get_dir
+local windower_path = windower.windower_path
+local     ffxi_path = windower.ffxi_path
+local      pol_path = windower.pol_path
+-- local launcher_path = windower.launcher_path 
+
+local dlls = { 'd3d8.dll', 'dxgi.dll', 'ddraw.dll', 'd3dimm.dll', 'd3d9.dll', '.*%.conf' }
+
+windower.register_event('addon command',function(arg)
+    local arg = arg:lower()
+
+    local windower_settings = windower.get_windower_settings()
+
+    local str = ''
+
+    str = str:append('\nWindower path: ' .. windower_path)
+    str = str:append('\n Launcher version: ' .. windower_settings.launcher_version)
+    str = str:append('\n Hook version: ' .. windower_settings.hook_version)
+    str = str:append('\nFFXI path: ' .. ffxi_path)
+    str = str:append('\n Client version: ' .. windower_settings.ffxi_version)
+
+    -- Look for cases of mismatched foldername\addonname.lua
+    local addons_path = windower_path .. 'addons\\'
+    local addons = get_dir(addons_path)
+    for _,name in pairs(addons) do
+        if name:endswith('-master') then
+            if not windower.file_exists('%s%s\\%s.lua':format(addons_path, name, name)) then
+                str = str:append('\nWrong folder name: ' .. name)
+            end
+        end
+    end
+
+    -- Look for graphics enhancement dlls and config files
+    for i, folder in pairs({ ffxi_path, pol_path, launcher_path }) do
+        local files = get_dir(folder)
+
+        for _, name in pairs(files) do
+            for _, pattern in pairs(dlls) do
+                local found = name:imatch(pattern)
+                if found then
+                    str = str:append('\n%s found in %s':format(found, folder))
+                end
+            end
+        end
+    end
+
+    log(str)
+
+    if arg == 'copy' or not windower.ffxi.get_info().logged_in then
+        windower.copy_to_clipboard(str:enclose('```'))
```

Comment:
I don't like treating login screen and in-game differently. I would either always copy to clipboard, even with no argument, or always respect the arguments, and then I'd probably allow something like
```
debuginfo copy|log
```
To display in log as well as copy. Could then also add a `console` and/or `file` option, for the people who are not logged in but don't want to copy to clipboard (I often don't want my clipboard messed up).

Reply:
Fair. I'll add a log|copy|file argument, which defaults to `log`. 
I do think that `log` should print to console if the character is not logged in (and thus the chatlog is unavailable), tho, what do you prefer?

Reply:
I don't think it's necessary, but I wouldn't mind if you implemented it. Should definitely be able to specify the console as well though, even if the chat log is available.

Reply:
Do you want the `console` flag to be exclusive to the `log` flag, or can they happen concurrently?

Reply:
I think all four should be treated equally. Ideally, I'd want all of this to work:
```
debug -- use some default, probably console, since it always works
debug log|copy
debug log|console|file|copy
```

But honestly, if it's too much work, just accepting one is fine, too. Was just an idea to allow multiple outputs. I'm not sure how often that will be used at all, so maybe it's not worth the effort.

Either way, `console` should be unrelated to `log`.

Reply:
I already made all flags able to work at the same time, the only doubt was if to make console and log exclusive to each other. In the end I did not, so you can specify all 4 flags and have a chorus of information thrown at you!